src/exprs.c

```diff
@@ -1015,15 +1021,18 @@ Obj ListExpr1 (
     else {
         list = NEW_PLIST( T_PLIST, len );
     }
-    SET_LEN_PLIST( list, len );
+    if (safe) {
+        SET_LEN_PLIST(list, 0);
+    }
+    else {
+        SET_LEN_PLIST(list, len);
+    }
```

Comment:
Could also use `SET_LEN_PLIST(list, tildeInUse ? 0 : len);` for compactness -- but of course it's also perfectly fine as it is :-).